build.gradle

```diff
@@ -28,72 +25,80 @@ allprojects {
     }
 }
 
+ext {
+    buildVersions = [
+            minSdk    : 16,
+            targetSdk : 27,
+            compileSdk: 27,
+            buildTools: "27.0.2"
+    ]
+
+    libsVersions = [
+            kotlin    : kotlin_version,
+            support   : "27.1.0",
+            constraint: "1.0.2",
+            picasso   : "2.5.2",
+            glide     : "4.6.1",
+            gson      : "2.8.2",
+            canarinho : "1.2.0",
+            lottie    : "2.5.0-rc1"
+    ]
+
+    testLibsVersions = [
+            jUnit        : "4.12",
+            espresso     : "3.0.1",
+            mockito      : "2.12.0",
+            mockitoKotlin: "1.5.0",
+            roboeletric  : "3.6.1"
+    ]
+
+    lintersVersions = [
+            ktlint: "0.19.0",
+            detekt: "1.0.0.RC6-3"
+    ]
+}
+
 subprojects {
-    apply from: rootProject.file("tools/jacoco.gradle")
     apply plugin: 'org.jetbrains.dokka'
-}
+    apply plugin: "jacoco"
 
-apply plugin: "jacoco"
+    apply from: "$rootDir/tools/quality.gradle"
+    apply from: "$rootDir/tools/jacoco.gradle"
+    apply from: "$rootDir/tools/detekt.gradle"
 
-jacoco {
-    toolVersion = "0.8.0"
-}
+    jacoco.toolVersion = "0.8.0"
 
-task jacocoRootReport(type: JacocoReport) {
-    group "Reporting"
-    description "Generate Jacoco coverage reports."
+    task jacocoRootReport(type: JacocoReport) {
```

Comment:
Are you sure this should go here (inside the subprojects closure)? This is not a module jacoco report but rather a merge of all coverage results.

Reply:
You're right! I went on a killing spree and shoot myself in the foot. Will remove this from here and leave it on the root project.

Reply:
hahaha no prob